semantics/Makefile

```diff
@@ -125,19 +126,17 @@ endif
 # See https://www.gnu.org/software/make/manual/html_node/Remaking-Makefiles.html
 Makefile: $(DEPENDS) ;
 
-
-# The included makefiles depend on these.
-K_FILE_DEPS := $(wildcard *.k)
-K_FILE_DEPS += $(shell find $(OUTPUT_DIR) -type f -name '*.k')
-K_FILE_DEPS += $(shell find $(PROFILE_DIR)/semantics -type f -name '*.k')
-
-$(BASIC_DEPENDS): $(GENERATED_MAKEFILE_DIR)/basic-%.dep.mk: $(K_FILE_DEPS) \
+$(BASIC_DEPENDS): $(GENERATED_MAKEFILE_DIR)/basic-%.dep.mk: %.k \
                                                             | $(GENERATED_MAKEFILE_DIR)
-	$(KDEP) $(KDEP_FLAGS) -d "$(OUTPUT_DIR)/$*-kompiled" -- $*.k > $@
+	$(KDEP) $(KDEP_FLAGS) -d "$(OUTPUT_DIR)/$*-kompiled" -- $*.k \
+		| perl -pe 's|(.*?)\s*:|\1 $@:|g' \
+	> $@
```

Comment:
The perl expression forces `$@` to depend on the same things that the targets inside `$@` depend on.

Reply:
But the dependencies themselves really need to be recalculated if any of the `.k` files change, right? I don't see how you can avoid that.

Reply:
That's exactly what this does. It makes the generated makefiles depend on the same things that the targets inside depend on. In other words, if any of the targets inside is out of date, so is the generated makefile.